### PR TITLE
Supporting template evaluation in ingress hosts

### PR DESCRIPTION
**Pull request by jkroepke**
#### Overview

Hi, I would like to use dex as part of a umbrella helm chart. For the user experience, there is currently global (sub)domain configuration across all helm charts and sub-charts are able to use the helm evaluation function to build a hostname. After merging the PR, I could use somelike like this in my value yaml

```yaml
dex:
  ingress:
    annotations:
      cert-manager.io/cluster-issuer: letsencrypt
    hosts:
    - host: 'auth.{{ $.Values.global.domain }}'
      paths:
      - path: /
         pathType: Prefix
    tls:
    - secretName: tls-secret-dex
      hosts:
      - 'auth.{{ $.Values.global.domain }}'
```

And user only needs to define `--set global.domain=example.com`.

Note, w/o this, end users needs to pass the whole list item, since helm is not able to merge lists.

#### Special notes for your reviewer

[kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/blob/c3b697d31b8631fb8847f7f313155deeb5962f98/charts/kube-prometheus-stack/templates/prometheus/ingress.yaml#L33) and grafana helm chart are supporting this scenario, too

#### Checklist

- [x] Change log updated in `Chart.yaml` (see the contributing guide for details)
- [x] Chart version bumped in `Chart.yaml` (see the contributing guide for details)
- [x] Documentation regenerated by running `make docs`

### CODE REVIEW

**Comment by jkroepke** [comment]
@sagikazarmark I'm aware pinging maintainers is not such a good thing! I would highly appreciate some feedback here.

**Comment by sagikazarmark** [comment]
@jkroepke I maintain dozens, if not hundreds of projects (which means I get at least a hundred emails from GitHub daily). Pinging me is the ONLY way to get my attention. Don't feel bad about that (but also don't take it personally if I don't answer for a long time).

I'm generally not a huge fan of this pattern, but it's a small change, low maintenance and may actually be more popular than I would have thought (didn't know prometheus uses this pattern), so I think it's okay to add.

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Please run `make docs`

**Comment by jkroepke** [comment]
Thanks @sagikazarmark, I additionally run `make docs`

**Comment by jkroepke** [comment]
@sagikazarmark are the actions failures related to this change?

**Comment by jkroepke** [comment]
@sagikazarmark is there anything what I can do to unblock this here?

**Comment by jkroepke** [comment]
@sagikazarmark how we can procceed here? Removing old kubernetes versions may be the best option here. (1.19+ is green)

**Comment by sagikazarmark** [comment]
Can you please rebase @jkroepke ?

**Comment by jkroepke** [comment]
@sagikazarmark rebased

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**
